lib/bindings/python/rust/lib.rs

```diff
@@ -417,17 +419,16 @@ impl Endpoint {
         &self,
         py: Python<'p>,
         generator: PyObject,
-        lease: Option<&PyLease>,
+        // Not used. We always attach to the primary lease.
+        // We can't rename or remove this because it might break Python.
+        #[allow(unused)] lease: Option<&PyLease>,
     ) -> PyResult<Bound<'p, PyAny>> {
         let engine = Arc::new(engine::PythonAsyncEngine::new(
             generator,
             self.event_loop.clone(),
         )?);
         let ingress = JsonServerStreamingIngress::for_engine(engine).map_err(to_pyerr)?;
-        let mut builder = self.inner.endpoint_builder().handler(ingress);
-        if lease.is_some() {
-            builder = builder.lease(lease.map(|l| l.inner.clone()));
-        }
+        let builder = self.inner.endpoint_builder().handler(ingress);
```

Comment:
Does this undo the "graceful shutdown feature" added via custom lease? CC @ishandhanani @tedzhouhk

Reply:
Yes. I _think_ they don't need it any more. Ishan can you confirm?